test/framework/filetools.py

```diff
@@ -355,6 +355,49 @@ def test_is_readable(self):
         os.chmod(test_file, 0)
         self.assertFalse(ft.is_readable(test_file))
 
+    def test_symlink_readlink_file(self):
+        """Test read link target file"""
+
+        # write_file and read_file tests are elsewhere. so not getting their states
+        fp = os.path.join(self.test_prefix, 'test.txt')
+        txt = "test_my_link_file"
+        ft.write_file(fp, txt)
+
+        link = os.path.join(self.test_prefix, 'test.link')
+        # creating the link file
+        ft.symlink(fp, link)
+
+        # checking if file is symlink
+        self.assertTrue(os.path.islink(link))
+        # reading link target and comparing to file name
+        self.assertEqual(os.path.realpath(fp), ft.readlink(link))
```

Comment:
hmm, I think this may fail, e.g. if `self.test_prefix` is a symlink...

`os.path.realpath` will give you the fully resolved location, while `ft.readlink` will only resolve one symlink 'level', I think?